### PR TITLE
ksmoothdock: init at 6.3 stable

### PR DESCRIPTION
**Pull request by SCOTT-HAMILTON**
###### Motivation for this change
Needed a nice dock for my MacOS KDE theming exerience. For the momen latte dock seems too buggy.

###### Things done
Packaged ksmoothdock at v6.2 stable

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SCOTT-HAMILTON** [comment]
The patch is currently discussed here https://github.com/dangvd/ksmoothdock/pull/117

**Comment by IvarWithoutBones** [comment]
meta doesn't seem to be matching the application

**Comment by SCOTT-HAMILTON** [comment]
Fixed here https://github.com/NixOS/nixpkgs/pull/88204/commits/ed9b1fb73c5759a2815258972e1965f15735a715

File: pkgs/desktops/plasma-5/ksmoothdock/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{ lib
```
Comment (ttuegel) [inline pkgs/desktops/plasma-5/ksmoothdock/default.nix:1]: This package isn't part of Plasma, so it should move to `pkgs/applications/misc/ksmoothdock`.

**Review by ttuegel — CHANGES_REQUESTED** [review]
This needs an entry in `pkgs/top-level/all-packages.nix`.

File: pkgs/desktops/plasma-5/ksmoothdock/default.nix (hunk 1 above)
Comment (SCOTT-HAMILTON) [reply pkgs/desktops/plasma-5/ksmoothdock/default.nix]: Moved to pkgs/applications/kde/ksmoothdock

Comment (SCOTT-HAMILTON) [reply pkgs/desktops/plasma-5/ksmoothdock/default.nix]: https://github.com/NixOS/nixpkgs/pull/88204/commits/4013b56c7dc6003548eb93cfd0af10fa1895fdcb

**Comment by IvarWithoutBones** [comment]
That commit just seems to remove the derivation all together?

**Comment by SCOTT-HAMILTON** [comment]
Asked cleanup done.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 88204 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary><br>- ksmoothdock</details>

**Pull request merged**